src/resources/Microsoft.Authorization/policyDefinitions/Deny-VMUnmanagedDisk.json

```diff
@@ -0,0 +1,67 @@
+{
+    "name": "Deny-VM-with-unmanaged-disk",
+    "apiVersion": "2021-06-01",
+    "scope": "null",
+    "properties": {
+        "policyType": "Custom",
+        "mode": "all",
+        "displayName": "Deny virtual machines and virtual machine scalesets not using managed disk",
+        "description": "Deny virtual machines not using managed disk. It checks the managedDisk property on virtual machine OS, and Datadisk fields.",
+        "metadata": {
+            "version": "1.0.0.0",
+            "category": "Compute",
+            "source": "https://github.com/Azure/Enterprise-Scale/",
+            "alzCloudEnvironments": [
+                "AzureCloud"
+            ]
+        },
+        "parameters": {
+            "effectType": {
+                "type": "string",
+                "defaultValue": "Deny",
+                "allowedValues": [
+                    "Audit",
+                    "Deny",
+                    "Disabled"
+                ],
+                "metadata": {
+                    "displayName": "Effect",
+                    "description": "Enable or disable the execution of the policy"
+                }
+            }
+        },
+        "policyRule": {
+            "if": {
+                "anyOf": [
+                    {
+                        "allOf": [
+                            {
+                                "field": "type",
+                                "equals": "Microsoft.Compute/virtualMachines"
+                            },
+                            {
+                                "field": "Microsoft.Compute/virtualMachines/storageProfile.osDisk.managedDisk",
+                                "exists": "false"
+                            }
+                        ]
+                    },
+                    {
+                        "allOf": [
+                            {
+                                "field": "type",
+                                "equals": "Microsoft.Compute/virtualMachineScaleSets"
+                            },
+                            {
+                                "field": "Microsoft.Compute/virtualMachineScaleSets/virtualMachineProfile.storageProfile.osDisk.managedDisk",
+                                "exists": "false"
+                            }
+                        ]
+                    }
+                ]
```

Comment:
@aarunraaj @Springstone what about data disks? This just covers OS disks?

Reply:
@jtracey93 , yes this covers the OS disk. The reason behind I selected this is (correct me if I am wrong, so I can update the data disk field), the unmanaged data disks cannot be attached on the VM which has managed disk, so this is out of scope in context. Secondly, when a VM attached an OS disk as unmanaged disk during creation (even if it has a data disk attached), the VM creation will be denied. For an existing VM this virtual machine will be a non-compliant resource. Let me know your thoughts.

Reply:
@jtracey93 @aarunraaj I tend to agree, I spent an hour trying to deploy an unmanaged disk and it's not easy. Essentially this is not restricted to deploying a VHD from a storage account. My understanding is also that you can not mix managed and unmanaged disk, which makes sense from an availability perspective. Blocking OS deployment should be sufficient I think

Reply:
If you are both happy then so am I. Can we please update the names and descriptions to reflect its only OS disks etc. though please. @aarunraaj @Springstone